components/slot-fill/slot.js

```diff
@@ -30,25 +36,40 @@ class Slot extends Component {
 
 		if ( this.props.name !== name ) {
 			unregisterSlot( this.props.name );
-			registerSlot( name, this.node );
+			registerSlot( name, this );
 		}
 	}
 
-	registerSlot( node ) {
-		const { registerSlot = noop } = this.context;
-
+	bindNode( node ) {
 		this.node = node;
-		registerSlot( this.props.name, node );
 	}
 
 	render() {
-		return <div ref={ this.registerSlot } />;
+		const { name, bubblesVirtually = false } = this.props;
+		const { getFills = noop } = this.context;
+
+		if ( bubblesVirtually ) {
+			return <div ref={ this.bindNode } />;
+		}
+
+		return (
+			<div ref={ this.bindNode }>
+				{ map( getFills( name ), ( fill ) => {
+					const fillKey = fill.props.instanceId;
+					return Children.map( fill.props.children, ( child, childIndex ) => {
+						const childKey = fillKey + '---' + ( child.props.key || childIndex );
```

Comment:
Very minor: maybe const childKey = `${ fillKey }---${ ( child.props.key || childIndex ) }`; because in most places we use this syntax.